core/domain/collection_services.py

```diff
@@ -312,6 +313,51 @@ def get_multiple_collections_by_id(collection_ids, strict=True):
     return result
 
 
+def get_collection_and_collection_rights_by_id(collection_id, version=None):
+    """Returns a tuple for collection domain object and collection rights
+        object.
+
+    Args:
+        collection_id: str. Id of the collection.
+        version: str. Version of the collection.
+
+    Returns:
+        Tuple(collection object or None, collection rights object or None).
+    """
+    collection_memcache_key = _get_collection_memcache_key(
+        collection_id, version=version)
+    memcached_collection = memcache_services.get_multi(
+        [collection_memcache_key]).get(collection_memcache_key)
+
+    if memcached_collection is not None:
+        collection_rights = rights_manager.get_collection_rights(
+            collection_id, strict=False)
+        return (memcached_collection, collection_rights)
+    else:
+        collection_and_rights = (
+            datastore_services.fetch_multiple_entities_by_ids_and_models(
+                [
+                    ('CollectionModel', [collection_id]),
+                    ('CollectionRightsModel', [collection_id])
+                ]))
+        if collection_and_rights[0][0]:
```

Comment:
Should this be: if collection_and_rights[0][0] is not None?

Ditto below for [1][0].

Reply:
ok.